app/lib/search/search_client.dart

```diff
@@ -37,7 +37,8 @@ class SearchClient {
     final String serviceUrl = '$httpHostPort/search$serviceUrlParams';
 
     Future<PackageSearchResult> searchFn() async {
-      final response = await getUrlWithRetry(_httpClient, serviceUrl);
+      final response = await getUrlWithRetry(_httpClient, serviceUrl,
+          timeout: Duration(seconds: 5));
```

Comment:
`getUrlWithRetry` doesn't seem like the best logic... but probably better than nothing.. ie. missing exponential backoff..

Reply:
yeah, agreed, but it doesn't immediately fit the `package:retry` model of retries... maybe a thing to refactor later...